src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -673,6 +682,18 @@ public static unsafe string Join(string? separator, string?[] value, int startIn
             }
         }
 
+        // Joins an array of strings together as one string with a separator between each original string.
+        //
+        private static unsafe string Join(string? separator, ReadOnlySpan<string?> value, int startIndex, int count)
```

Comment:
Why does this need to take a startIndex and count? Is it not always 0 and value.Length?

Reply:
Was following the pattern of array above, removed these parameters and passed down 0 and value.Length